web/frontend/src/types/configuration.ts

```diff
@@ -0,0 +1,99 @@
+export type ID = string;
+
+export const RANK = 'rank';
+export const SELECT = 'select';
+export const SUBJECT = 'subject';
+export const TEXT = 'text';
+export const ROOT_ID: ID = '0';
+
+// Rank describes a "rank" question, which requires the user to rank choices.
+export interface RankQuestion {
+  ID: ID;
+  Title: string;
+  MaxN: number;
+  MinN: number;
+  Choices: Array<string>;
+}
+
+// Text describes a "text" question, which allows the user to enter free text.
+export interface TextQuestion {
+  ID: ID;
+  Title: string;
+  MaxN: number;
+  MinN: number;
+  MaxLength: number;
+  Regex?: string;
+  Choices: Array<string>;
+}
+
+// Select describes a "select" question, which requires the user to select one
+// or multiple choices.
+export interface SelectQuestion {
+  ID: ID;
+  Title: string;
+  MaxN: number;
+  MinN: number;
+  Choices: Array<string>;
+}
+
+export interface Subject {
+  ID: ID;
+  Title: string;
+  // Order defines the order of the different question, which all have a unique
+  // identifier. This is purely for display purpose.
+  Order: Array<ID>;
+  Subjects?: Array<Subject>;
+  Selects?: Array<SelectQuestion>;
+  Ranks?: Array<RankQuestion>;
+  Texts?: Array<TextQuestion>;
+}
+
+// Configuration contains the configuration of a new poll.
+export interface Configuration {
+  MainTitle: string;
+  Scaffold: Array<Subject>;
+}
+
+export interface Question {
```

Comment:
This is misleading, as a `Subject` is not a question. Should better be like `SubjectElement`, or `FormElement`.